Formula/gromacs.rb

```diff
@@ -50,11 +51,15 @@ def install
 
     inreplace "src/gromacs/gromacs-config.cmake.cmakein", "@GROMACS_CXX_COMPILER@", cxx
 
-    mkdir "build" do
-      system "cmake", "..", *std_cmake_args, "-DGROMACS_CXX_COMPILER=#{cxx}",
-                                             "-DGMX_VERSION_STRING_OF_FORK=#{tap.user}"
-      system "make", "install"
-    end
+    args = %W[
+      -DGROMACS_CXX_COMPILER=#{cxx}
+      -DGMX_VERSION_STRING_OF_FORK=#{tap.user}
+    ]
+    # Force SSE2/SSE4.1 for compatibility when building Intel bottles
+    args << "-DGMX_SIMD=#{MacOS.version.requires_sse41? ? "SSE4.1" : "SSE2"}" if Hardware::CPU.intel? && build.bottle?
```

Comment:
When I last looked into this I recall reading documentation that suggested this actually does runtime detection of CPU features. If so we might be breaking that with our flag refurbishment.

Reply:
Other package managers seem to set this so if it does have runtime detection, I'd guess it either only covers part of it or isn't sufficient in some cases.

Reply:
The `INSTALL` mentions:
```
Portability aspects
~~~~~~~~~~~~~~~~~~~

A GROMACS build will normally not be portable, not even across
hardware with the same base instruction set, like x86. Non-portable
hardware-specific optimizations are selected at configure-time, such
as the SIMD instruction set used in the compute kernels. This
selection will be done by the build system based on the capabilities
of the build host machine or otherwise specified to "cmake" during
configuration.

Often it is possible to ensure portability by choosing the least
common denominator of SIMD support, e.g. SSE2 for x86. In rare cases
of very old x86 machines, ensure that you use "cmake
-DGMX_USE_RDTSCP=off" if any of the target CPU architectures does not
support the "RDTSCP" instruction.  However, we discourage attempts to
use a single GROMACS installation when the execution environment is
heterogeneous, such as a mix of AVX and earlier hardware, because this
will lead to programs (especially mdrun) that run slowly on the new
hardware. Building two full installations and locally managing how to
call the correct one (e.g. using a module system) is the recommended
approach. Alternatively, one can use different suffixes to install
several versions of GROMACS in the same location. To achieve this, one
can first build a full installation with the least-common-denominator
SIMD instruction set, e.g. "-DGMX_SIMD=SSE2", in order for simple
commands like "gmx grompp" to work on all machines, then build
specialized "gmx" binaries for each architecture present in the
heterogeneous environment. By using custom binary and library suffixes
(with CMake variables "-DGMX_BINARY_SUFFIX=xxx" and
"-DGMX_LIBS_SUFFIX=xxx"), these can be installed to the same location.
```